homeassistant/components/dwd_weather_warnings/config_flow.py

```diff
@@ -27,27 +35,58 @@ async def async_step_user(
-            if not await self.hass.async_add_executor_job(
-                DwdWeatherWarningsAPI, region_identifier
-            ):
-                errors["base"] = "invalid_identifier"
-
-            if not errors:
-                # Set the unique ID for this config entry.
-                await self.async_set_unique_id(region_identifier)
-                self._abort_if_unique_id_configured()
-
-                return self.async_create_entry(title=region_identifier, data=user_input)
+            valid_options = (CONF_REGION_IDENTIFIER, CONF_GPS_TRACKER)
+
+            # Check, if either CONF_REGION_IDENTIFIER or CONF_GPS_TRACKER has been set.
+            if all(k not in user_input for k in valid_options):
+                errors["base"] = "no_identifier"
+            elif all(k in user_input for k in valid_options):
+                errors["base"] = "ambiguous_identifier"
+            elif CONF_REGION_IDENTIFIER in user_input:
+                # Validate region identifier using the API
+                identifier = user_input[CONF_REGION_IDENTIFIER]
+
+                if not await self.hass.async_add_executor_job(
+                    DwdWeatherWarningsAPI, identifier
+                ):
+                    errors["base"] = "invalid_identifier"
+
+                if not errors:
+                    # Set the unique ID for this config entry.
+                    await self.async_set_unique_id(identifier)
+                    self._abort_if_unique_id_configured()
+
+                    return self.async_create_entry(title=identifier, data=user_input)
+            elif CONF_GPS_TRACKER in user_input:
+                # Validate position using the API
+                device_tracker = user_input[CONF_GPS_TRACKER]
+                position = get_position_data(self.hass, device_tracker)
+
+                if not await self.hass.async_add_executor_job(
+                    DwdWeatherWarningsAPI, position
+                ):
+                    errors["base"] = "invalid_identifier"
+
+                # Position is valid here, because the API call was successful.
+                if not errors and position is not None:
+                    # Set the unique ID for this config entry.
+                    await self.async_set_unique_id(f"{position[0]}-{position[1]}")
+                    self._abort_if_unique_id_configured()
+
+                    return self.async_create_entry(
+                        title=device_tracker.removeprefix("device_tracker."),
+                        data=user_input,
+                    )
 
         return self.async_show_form(
             step_id="user",
             errors=errors,
             data_schema=vol.Schema(
                 {
-                    vol.Required(CONF_REGION_IDENTIFIER): cv.string,
+                    vol.Optional(CONF_REGION_IDENTIFIER): cv.string,
+                    vol.Optional(CONF_GPS_TRACKER): EntitySelector(
+                        EntitySelectorConfig(domain="device_tracker")
```

Comment:
We don't allow device or service integrating integrations to use other entities than their own entities. That kind of interface is reserved for integrations that should consume other entities, like template integrations and similar integrations.

Reply:
Oh, that's new to me. Back when I first implemented this feature (had to split up the changes because it was too much) it seemed to fine.

Thanks for asking around

Reply:
Hey @MartinHjelmare, since it's been a while, I wanted to ask if you had the time to check on this matter

Reply:
The conclusion is that the approach in this PR is OK, we will allow integrations to use entities from other integrations which have a location, e.g. zone and device_tracker entities can be used in other integrations.

Reply:
Thank you! I'll look over everything as soon as I get the time